sdk/src/store.rs

```diff
@@ -6209,6 +6213,160 @@ pub mod tests {
+
+        // should be an update manifest
+        assert!(um.update_manifest());
+
+        // should have valid bmff hash binding
+        assert!(um_report.has_status(validation_status::ASSERTION_BMFFHASH_MATCH));
+
+        // now add a new oridinary manifest to restore the original manifest (collapsed manifest)
+
+        // create a new ordinary claim
+        let mut claim2 = Claim::new("adobe unit test", Some("ordinary_manifest_vendor"), 2);
+        // ClaimGeneratorInfo is mandatory in Claim V2
+        let cgi = ClaimGeneratorInfo::new("claim_v2_unit_test");
+        claim2.add_claim_generator_info(cgi);
+
+        // make update PC claim the parent of the ordinary claim
+        let update_pc = um_store.provenance_claim().unwrap();
+        let ingredient_hashes = um_store.get_manifest_box_hashes(&update_pc);
+        let parent_hashed_uri = HashedUri::new(
+            um_store.provenance_path().unwrap(),
+            Some(update_pc.alg().to_string()),
+            &ingredient_hashes.manifest_box_hash,
+        );
+        let signature_hashed_uri = HashedUri::new(
+            to_signature_uri(update_pc.label()),
+            Some(update_pc.alg().to_string()),
+            &ingredient_hashes.signature_box_hash,
+        );
+
+        let validation_results = ValidationResults::from_store(&um_store, &report);
+
+        let ingredient = Ingredient::new_v3(Relationship::ParentOf)
+            .set_active_manifests_and_signature_from_hashed_uri(
+                Some(parent_hashed_uri),
+                Some(signature_hashed_uri),
+            ) // mandatory for v3
+            .set_validation_results(Some(validation_results)); // mandatory for v3
+
+        claim2.add_assertion(&ingredient).unwrap();
+
+        // create mandatory opened action
+        let ingredient = claim2.ingredient_assertions()[0];
+        let ingregient_uri = to_assertion_uri(claim2.label(), &ingredient.label());
+        let ingredient_hashed_uri = HashedUri::new(
+            ingregient_uri,
+            Some(claim2.alg().to_owned()),
+            ingredient.hash(),
+        );
+
+        let opened = Action::new("c2pa.opened")
+            .set_parameter("ingredients", vec![ingredient_hashed_uri])
+            .unwrap();
+        let editted = Action::new("c2pa.edited");
+        let actions = Actions::new().add_action(opened).add_action(editted);
+
+        // add action
+        claim2.add_assertion(&actions).unwrap();
+
+        um_store.commit_claim(claim2).unwrap();
+        um_store.save_to_asset(&op, signer.as_ref(), &op2).unwrap();
+
+        let mut collapsed_report =
+            StatusTracker::with_error_behavior(ErrorBehavior::StopOnFirstError);
+
+        // read back in store with update manifest
+        let collapsed_store =
+            Store::load_from_asset(op2.as_path(), true, &mut collapsed_report).unwrap();
+
+        let cm = collapsed_store.provenance_claim().unwrap();
+        assert!(!cm.update_manifest());
```

Comment:
With this case, the update manifest still exists in the store just not as the provenance claim?